qa/tasks/cephfs/test_misc.py

```diff
@@ -142,6 +142,36 @@ def test_evict_client(self):
         ls_data = self.fs.mds_asok(['session', 'ls'])
         self.assert_session_count(1, ls_data)
 
+    def test_dont_mark_unresponsive_client_stale(self):
+        """
+        Test that an unresponsive client holding no caps is evicted directly,
+        without being marked as stale even after session_timeout, at
+        session_autoclose.
+
+        To conduct this test we need more that two clients since a single
+        client is never evcited by MDS.
+        """
+        # setup part for the test
+        SESSION_TIMEOUT = 30
+        SESSION_AUTOCLOSE = 50
+        mount_a_gid = self.mount_a.get_global_id()
+        mount_a_pid = self.mount_a.get_client_pid()
+        self.fs.set_var('session_timeout', SESSION_TIMEOUT)
+        self.fs.set_var('session_autoclose', SESSION_AUTOCLOSE)
+        self.assert_session_count(2, self.fs.mds_asok(['session', 'ls']))
+
+        self.mount_a.run_shell(['mkdir', 'dir'])
+        self.mount_a.send_signal('sigstop')
+        time.sleep(SESSION_TIMEOUT + 2)
+        self.assert_session_state(mount_a_gid, "open")
+
+        self.mount_b.run_shell(['stat', 'dir'])
+        time.sleep(SESSION_AUTOCLOSE - SESSION_TIMEOUT + 2)
+        self.assert_session_count(1, self.fs.mds_asok(['session', 'ls']))
+
+        # Kill the corresponding ceph-fuse process
+        self.mount_b.run_shell(['kill', '-sigkill', mount_a_pid])
+
```

Comment:
I think it's better to move this test in test_client_recovery

Reply:
Done.